src/Operations.jl

```diff
@@ -597,22 +598,36 @@ end
 function download_artifacts(pkg_roots::Vector{String}; platform::AbstractPlatform=HostPlatform(),
                             verbose::Bool=false)
     # List of Artifacts.toml files that we're going to download from
-    artifacts_tomls = String[]
+    artifacts_tomls = Tuple{String,Dict}[]
 
     for path in pkg_roots
         # Check to see if this package has an (Julia)Artifacts.toml
         for f in artifact_names
             artifacts_toml = joinpath(path, f)
             if isfile(artifacts_toml)
-                push!(artifacts_tomls, artifacts_toml)
+                hook_path = joinpath(path, ".pkg", "select_artifacts_hook.jl")
+
+                if isfile(hook_path)
+                    mktempdir() do project
+                        meta_toml = String(read(`$(Base.julia_cmd()) --project=$(project) $(hook_path)`))
+                        push!(artifacts_tomls, (artifacts_toml, TOML.parse(meta_toml)))
```

Comment:
```suggestion
                        push!(artifacts_tomls, (artifacts_toml, Base.parsed_toml(meta_toml)))
```

if that is ok.

Reply:
Should be fine; that file is only ever read once, since it's a temporary file and immediately deleted.

Reply:
Actually, I lied.  The `meta_toml` is the TOML content itself, so there's really no benefit to using `Base.parsed_toml` here.